p2p/p2p.go

```diff
@@ -378,6 +471,61 @@ func (s *Service) Peers() []string {
 	return PeerIdToStringArray(s.host.Network().Peers())
 }
 
+// TODO: message handling
+func (s *Service) handleBroadcastStream(stream net.Stream) {
+
+	log.Debug("got stream", "peer", stream.Conn().RemotePeer())
+
+	rw := bufio.NewReadWriter(bufio.NewReader(stream), bufio.NewWriter(stream))
+	lengthByte, err := rw.Reader.ReadByte()
+	if err != nil {
+		log.Error("failed to read message length", "peer", stream.Conn().RemotePeer(), "error", err)
+		return
+	}
+
+	// decode message length using LEB128
+	length := LEB128ToUint64([]byte{lengthByte})
+
+	// read message type byte
+	msgType, err := rw.Reader.Peek(1)
+	if err != nil {
+		log.Error("failed to read message type", "err", err)
+		return
+	}
+
+	// read entire message
+	rawMsg, err := rw.Reader.Peek(int(length) - 1)
+	if err != nil {
+		log.Error("failed to read message", "err", err)
+		return
+	}
+
+	log.Debug("got stream", "peer", stream.Conn().RemotePeer(), "msg", fmt.Sprintf("0x%x", rawMsg))
+
+	// decode message
+	msg, err := DecodeMessage(rw.Reader)
+	if err != nil {
+		log.Error("failed to decode message", "error", err)
+		return
+	}
+
+	log.Debug("got message", "peer", stream.Conn().RemotePeer(), "type", msgType, "msg", msg.String())
+
+	err = s.Broadcast(msg)
```

Comment:
not sure what the difference in between this and the other handle stream function apart from the broadcasting, you should remove this and add logic to handle broadcasting for different message types inside the existing stream handler